source/agora/common/Deserializer.d

```diff
@@ -136,8 +136,13 @@ unittest
 
 public T deserializeFull (T) (scope ubyte[] data) @safe
 {
+    import std.format;
     scope DeserializeDg dg = (size) @safe
     {
+        if (size > data.length)
+            throw new Exception(
```

Comment:
So the idea is that deserialization errors are recoverable. I guess it depends entirely on the calling code, so that would make sense..

Reply:
They should be recoverable (because the only to proof that a given `ubyte[]` is valid is to feed it to the deserializer), but in this case it was to get a better error message.

Reply:
Should we use a reusable exception? Because `new`-ing here is also a forum of DDoS I would think.. although the GC should just collect the throwaway exceptions I suppose..

Reply:
Anyway we can move that into a separate PR.